File: apps-src/apps/projectfiles/windows-prj/template/window/simple.lua
-- var.
aplt_leagor_studio__simple = {
	var = {},
};

function aplt_leagor_studio__simple.construct()
	local var = aplt_leagor_studio__simple.var;
	var.startup_ticks_ = rose.get_ticks();
	return 0;
end

function aplt_leagor_studio__simple.pre_show(dlg, window)
	local this = aplt_leagor_studio__simple;
	local var = this.var;
	local aplt = aplt_leagor_studio;
	local _ = aplt_leagor_studio._

	var.dlg_ = dlg;
	var.window_ = window;

	window:set_label("misc/white_background.png");

	gui2.find_widget(window, "ok", false, true):set_visible(gui2.tvisible_INVISIBLE);

end

function aplt_leagor_studio__simple.post_show()
	aplt_leagor_studio__simple.var = {};
	-- rose.cpp_breakpoint();
end

function aplt_leagor_studio__simple.timer_handler(now)
	local this = aplt_leagor_studio__simple;
	local var = this.var;
	local _ = aplt_leagor_studio._
end

File: apps-src/apps/projectfiles/windows-prj/template/window/simple.cpp
#define GETTEXT_DOMAIN "studio-lib"

#include "gui/dialogs/simple.hpp"

#include "gui/widgets/label.hpp"
#include "gui/widgets/window.hpp"
#include "gettext.hpp"

namespace gui2 {

REGISTER_DIALOG(studio, simple)

tsimple::tsimple()
{
}

void tsimple::pre_show()
{
	window_->set_label("misc/white_background.png");
	
	find_widget<tlabel>(window_, "title", false).set_label(_("Hello World"));
}

void tsimple::post_show()
{
}

} // namespace gui2

